### PR TITLE
BHoM: Created ISpecification interface

### LINKED ISSUE #1144: BHoM: Create ISpecification interface
#### Description:

Will be used for CIH and specification definition, which will be central concepts in BHoM.

### PR DESCRIPTION
**Pull request by alelom**
### Issues addressed by this PR

Closes #1144

The specification interface defines a soon-to-be core concept in BHoM used for [object specification](https://github.com/CIHDigitalConfigurator/CIH_Toolkit) definition. 

### Test files

Not needed.

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
@alelom don't forget which oM is being modified for the change log please :smile:

**Review by al-fisher — APPROVED** [review]
LGTM

@alelom I have raised an issue here: https://github.com/BHoM/BHoM/issues/1152 for others to see what might be planned later - i.e. porting more functionality over to BHoM once prototyping is complete

File: BHoM/Interface/ISpecification.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.oM.Base
+{
+    public interface ISpecification : IObject
```
Comment (IsakNaslundBh) [inline BHoM/Interface/ISpecification.cs:29]: After discussion this morning, is it worth adding a breif description of what this is meant to be used for? @alelom

Comment (al-fisher) [reply BHoM/Interface/ISpecification.cs:29]: @IsakNaslundBh I have raised a specific issue on this already https://github.com/BHoM/BHoM/issues/1152

**Pull request merged**